client/consensus/Cargo.toml

```diff
@@ -3,27 +3,22 @@ name = "fc-consensus"
 version = "2.0.0-dev"
 authors = ["Parity Technologies <admin@parity.io>"]
 description = "Frontier consensus for substrate"
-edition = "2018"
+edition = "2021"
 license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 repository = "https://github.com/paritytech/frontier/"
 
 [dependencies]
-codec = { package = "parity-scale-codec", version = "2.0.0", features = ["derive"] }
-sp-core = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
-sp-blockchain = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
-sp-runtime = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
-sp-api = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
-sc-client-api = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
-sp-block-builder = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
-sp-inherents = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
+async-trait = "0.1"
+thiserror = "1.0"
+
+sp-api = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.16" }
```

Comment:
```suggestion
sp-api = { version = "4.0.0-dev", git = "https://github.com/paritytech/substrate" }
```

Please keep this reference to master branch for now. You can do `cargo update -p sp-io --precise <rev>` to point the lock file to the exact commit closest to v0.9.16. I understand this is not a great situation, but it'll otherwise complicate resolution when other repos refer to Substrate repo.

If you then want to include Frontier directly without forking and using a specific branch, you can use `[patch]` in your top-level workspace. See https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html

Reply:
Okay, I will revert the commit [74e3af6](https://github.com/paritytech/frontier/pull/583/commits/74e3af6437bcba32b56dcc8f26c4b748c27acbd9) and [6a4e334](https://github.com/paritytech/frontier/pull/583/commits/6a4e3344a59958b3dc7a1a9a75e65e3f3ef2366a), the commit [b341c2e](https://github.com/paritytech/frontier/pull/583/commits/b341c2e546cb2593c3ff2c1e6e5d1b5daceefe2a) uses the master branch